bindings/python/src/ser.rs

```diff
@@ -146,19 +149,30 @@ impl Serialize for SerializePyObject {
                     map.end()
                 }
             }
-            ObjectType::List => {
+            ObjectType::Tuple | ObjectType::List => {
```

Comment:
We can avoid the `match` statements below (with their unreachable `_` arms) by separating these two arms - the code is somewhat similar, but it could be solved by a macro. This way we'll avoid the performance penalty of checking the type of each list / tuple multiple times.

Reply:
I had this originally as two separate Matches, but wanted to reduce the code duplication, I can't imagine the match function is super slow in here, I'm assuming the python calls will be the slowest part here by a large margin

I've never written a rust macro, so will do some reading on how they work.

Reply:
Agree, the overhead should be relatively small here, but:

- It scales together with the number of lists & number of elements in each list, so it would be more visible with larger list/tuple heavy inputs;
- I assume that it also affects cache locality a bit as we always need to keep one more value around (`self.object_type`)
- It forces us to write code for unreachable cases;

The performance points need some profiling of course but from the top level, I prefer to avoid performance costs that have a relatively straightforward solution.

I am totally ok to have a bit of code duplication here as the blocks are relatively small.

Reply:
I did have a read through some tutorials on macros, but am a little lost on how to actually implement a macro here.

I could just copy and paste the code into match blocks if that would work for you, or if you have any input on how to make a macro for this, I'd be happy to update the PR.

Reply:
> I could just copy and paste the code into match blocks if that would work for you, or if you have any input on how to make a macro for this, I'd be happy to update the PR.

Absolutely! We can think about a macro later

Reply:
Awesome, I have added a commit that does this 

https://github.com/Stranger6667/jsonschema-rs/pull/312/commits/0ada8f37dfb0bd7d7ff9bc11ba390fffa4c1dfe7